// sanity-plugin-dashboard-widget-vercel/lib/components/DeploymentTarget/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _icons = require("@sanity/icons");

var _ui = require("@sanity/ui");

var _react = _interopRequireDefault(require("react"));

var _Deployments = _interopRequireDefault(require("../Deployments"));

var DeploymentTarget = function DeploymentTarget(props) {
  var item = props.item,
      onDialogEdit = props.onDialogEdit;
  var deploymentTarget = {
    deployHook: item.deployHook,
    deployLimit: item.deployLimit,
    name: item.name,
    projectId: item.projectId,
    teamId: item.teamId,
    token: item.token
  };
  return /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    style: {
      position: 'relative'
    }
  }, /*#__PURE__*/_react["default"].createElement(_ui.Flex, {
    align: "center",
    justify: "space-between",
    marginTop: 2,
    paddingX: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 2
  }, item.name), /*#__PURE__*/_react["default"].createElement(_ui.Tooltip, {
    content: /*#__PURE__*/_react["default"].createElement(_ui.Box, {
      padding: 2
    }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
      muted: true,
      size: 1
    }, "Edit deployment target")),
    placement: "left"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Button, {
    fontSize: 1,
    icon: _icons.EditIcon,
    mode: "bleed",
    onClick: function onClick() {
      return onDialogEdit(item);
    }
  }))), /*#__PURE__*/_react["default"].createElement(_Deployments["default"], {
    deploymentTarget: deploymentTarget
  }));
};

var _default = DeploymentTarget;
exports["default"] = _default;
