// sanity-plugin-dashboard-widget-vercel/lib/components/DialogForm/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _yup = require("@hookform/resolvers/yup");

var _ui = require("@sanity/ui");

var _uuid = require("@sanity/uuid");

var _react = require("@xstate/react");

var _react2 = _interopRequireDefault(require("react"));

var _reactHookForm = require("react-hook-form");

var yup = _interopRequireWildcard(require("yup"));

var _client = require("../../client");

var _constants = require("../../constants");

var _form = _interopRequireDefault(require("../../machines/form"));

var _sanitizeFormData = _interopRequireDefault(require("../../utils/sanitizeFormData"));

var _FormFieldInputText = _interopRequireDefault(require("../FormFieldInputText"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var formSchema = yup.object().shape({
  deployHook: yup.string().url('Deploy hook must be a valid URL'),
  deployLimit: yup.number().positive().integer().min(1, 'Deploy limit must no less than 1').max(15, 'Deploy limit must no higher than 15').typeError('Deploy limit must be a number').required('Deploy limit must be a positive integer between 1 and 15'),
  name: yup.string().required('Name cannot be empty'),
  projectId: yup.string().required('Vercel Project ID cannot be empty'),
  teamId: yup.string(),
  token: yup.string().required('Vercel Account Token cannot be empty')
});

var DialogForm = function DialogForm(props) {
  var deploymentTarget = props.deploymentTarget,
      onClose = props.onClose,
      onCreate = props.onCreate,
      onDelete = props.onDelete,
      onUpdate = props.onUpdate; // xstate

  var _useMachine = (0, _react.useMachine)(_form["default"], {
    services: {
      formSubmittedService: function () {
        var _formSubmittedService = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
          return _regenerator["default"].wrap(function _callee$(_context2) {
            while (1) {
              switch (_context2.prev = _context2.next) {
                case 0:
                  onClose();

                case 1:
                case "end":
                  return _context2.stop();
              }
            }
          }, _callee);
        }));

        function formSubmittedService() {
          return _formSubmittedService.apply(this, arguments);
        }

        return formSubmittedService;
      }(),
      // TODO: refactor
      createDocumentService: function () {
        var _createDocumentService = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(_context, event) {
          var document;
          return _regenerator["default"].wrap(function _callee2$(_context3) {
            while (1) {
              switch (_context3.prev = _context3.next) {
                case 0:
                  _context3.prev = 0;
                  _context3.next = 3;
                  return _client.client.create(_objectSpread({
                    _id: "vercel.".concat((0, _uuid.uuid)()),
                    _type: _constants.DEPLOYMENT_TARGET_DOCUMENT_TYPE
                  }, event.formData));

                case 3:
                  document = _context3.sent;

                  if (onCreate) {
                    onCreate(document);
                  }

                  return _context3.abrupt("return", Promise.resolve());

                case 8:
                  _context3.prev = 8;
                  _context3.t0 = _context3["catch"](0);
                  return _context3.abrupt("return", Promise.reject(_context3.t0));

                case 11:
                case "end":
                  return _context3.stop();
              }
            }
          }, _callee2, null, [[0, 8]]);
        }));

        function createDocumentService(_x, _x2) {
          return _createDocumentService.apply(this, arguments);
        }

        return createDocumentService;
      }(),
      // TODO: refactor
      deleteDocumentService: function () {
        var _deleteDocumentService = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3() {
          return _regenerator["default"].wrap(function _callee3$(_context4) {
            while (1) {
              switch (_context4.prev = _context4.next) {
                case 0:
                  if (!deploymentTarget) {
                    _context4.next = 11;
                    break;
                  }

                  _context4.prev = 1;
                  _context4.next = 4;
                  return _client.client["delete"](deploymentTarget._id);

                case 4:
                  if (onDelete) {
                    onDelete(deploymentTarget._id);
                  }

                  return _context4.abrupt("return", Promise.resolve());

                case 8:
                  _context4.prev = 8;
                  _context4.t0 = _context4["catch"](1);
                  return _context4.abrupt("return", Promise.reject(_context4.t0));

                case 11:
                case "end":
                  return _context4.stop();
              }
            }
          }, _callee3, null, [[1, 8]]);
        }));

        function deleteDocumentService() {
          return _deleteDocumentService.apply(this, arguments);
        }

        return deleteDocumentService;
      }(),
      // TODO: refactor
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      updateDocumentService: function () {
        var _updateDocumentService = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(_context, event) {
          var document;
          return _regenerator["default"].wrap(function _callee4$(_context5) {
            while (1) {
              switch (_context5.prev = _context5.next) {
                case 0:
                  if (!deploymentTarget) {
                    _context5.next = 12;
                    break;
                  }

                  _context5.prev = 1;
                  _context5.next = 4;
                  return _client.client.patch(deploymentTarget._id).set(event.formData).commit();

                case 4:
                  document = _context5.sent;

                  if (onUpdate) {
                    onUpdate(document);
                  }

                  return _context5.abrupt("return", Promise.resolve());

                case 9:
                  _context5.prev = 9;
                  _context5.t0 = _context5["catch"](1);
                  return _context5.abrupt("return", Promise.reject(_context5.t0));

                case 12:
                case "end":
                  return _context5.stop();
              }
            }
          }, _callee4, null, [[1, 9]]);
        }));

        function updateDocumentService(_x3, _x4) {
          return _updateDocumentService.apply(this, arguments);
        }

        return updateDocumentService;
      }()
    }
  }),
      _useMachine2 = (0, _slicedToArray2["default"])(_useMachine, 2),
      formState = _useMachine2[0],
      formStateTransition = _useMachine2[1];

  var formUpdating = formState.matches('creating') || formState.matches('deleting') || formState.matches('updating'); // react-hook-form

  var _useForm = (0, _reactHookForm.useForm)({
    defaultValues: {
      deployHook: (deploymentTarget === null || deploymentTarget === void 0 ? void 0 : deploymentTarget.deployHook) || '',
      deployLimit: (deploymentTarget === null || deploymentTarget === void 0 ? void 0 : deploymentTarget.deployLimit) || 5,
      name: deploymentTarget === null || deploymentTarget === void 0 ? void 0 : deploymentTarget.name,
      projectId: deploymentTarget === null || deploymentTarget === void 0 ? void 0 : deploymentTarget.projectId,
      teamId: (deploymentTarget === null || deploymentTarget === void 0 ? void 0 : deploymentTarget.teamId) || '',
      token: deploymentTarget === null || deploymentTarget === void 0 ? void 0 : deploymentTarget.token
    },
    mode: 'onChange',
    resolver: (0, _yup.yupResolver)(formSchema)
  }),
      _useForm$formState = _useForm.formState,
      errors = _useForm$formState.errors,
      isDirty = _useForm$formState.isDirty,
      isValid = _useForm$formState.isValid,
      handleSubmit = _useForm.handleSubmit,
      register = _useForm.register; // Callbacks
  // - submit react-hook-form


  var onSubmit = /*#__PURE__*/function () {
    var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(formData) {
      var sanitizedFormData;
      return _regenerator["default"].wrap(function _callee5$(_context6) {
        while (1) {
          switch (_context6.prev = _context6.next) {
            case 0:
              sanitizedFormData = (0, _sanitizeFormData["default"])(formData);
              formStateTransition(deploymentTarget ? 'UPDATE' : 'CREATE', {
                formData: sanitizedFormData
              });

            case 2:
            case "end":
              return _context6.stop();
          }
        }
      }, _callee5);
    }));

    return function onSubmit(_x5) {
      return _ref.apply(this, arguments);
    };
  }();

  var handleDelete = function handleDelete() {
    formStateTransition('DELETE', {
      id: deploymentTarget === null || deploymentTarget === void 0 ? void 0 : deploymentTarget._id
    });
  };

  var Footer = function Footer() {
    return /*#__PURE__*/_react2["default"].createElement(_ui.Box, {
      padding: 3
    }, /*#__PURE__*/_react2["default"].createElement(_ui.Flex, {
      justify: deploymentTarget ? 'space-between' : 'flex-end'
    }, deploymentTarget && /*#__PURE__*/_react2["default"].createElement(_ui.Button, {
      disabled: formUpdating,
      fontSize: 1,
      mode: "bleed",
      onClick: handleDelete,
      text: "Delete",
      tone: "critical"
    }), /*#__PURE__*/_react2["default"].createElement(_ui.Button, {
      disabled: formUpdating || !isDirty || !isValid,
      fontSize: 1,
      onClick: handleSubmit(onSubmit),
      text: deploymentTarget ? 'Update and close' : 'Create',
      tone: "primary"
    })));
  };

  return /*#__PURE__*/_react2["default"].createElement(_ui.Dialog, {
    footer: /*#__PURE__*/_react2["default"].createElement(Footer, null),
    header: "".concat(deploymentTarget ? 'Edit' : 'Create', " deployment target"),
    id: "create",
    onClose: onClose,
    width: 1,
    zOffset: _constants.Z_INDEX_DIALOG
  }, /*#__PURE__*/_react2["default"].createElement(_ui.Box, {
    as: "form",
    padding: 4,
    onSubmit: handleSubmit(onSubmit)
  }, /*#__PURE__*/_react2["default"].createElement("button", {
    style: {
      display: 'none'
    },
    tabIndex: -1,
    type: "submit"
  }), /*#__PURE__*/_react2["default"].createElement(_ui.Stack, {
    space: 5
  }, /*#__PURE__*/_react2["default"].createElement(_FormFieldInputText["default"], {
    disabled: formUpdating,
    description: "Name displayed in this plugin (e.g. production, staging)",
    error: errors === null || errors === void 0 ? void 0 : errors.name,
    label: "Name",
    name: "name",
    ref: register
  }), /*#__PURE__*/_react2["default"].createElement(_FormFieldInputText["default"], {
    disabled: formUpdating,
    error: errors === null || errors === void 0 ? void 0 : errors.token,
    label: "Vercel Account Token",
    name: "token",
    ref: register
  }), /*#__PURE__*/_react2["default"].createElement(_FormFieldInputText["default"], {
    disabled: formUpdating,
    error: errors === null || errors === void 0 ? void 0 : errors.projectId,
    label: "Vercel Project ID",
    name: "projectId",
    ref: register
  }), /*#__PURE__*/_react2["default"].createElement(_FormFieldInputText["default"], {
    description: "Required only if your project is owned by a team account",
    disabled: formUpdating,
    error: errors === null || errors === void 0 ? void 0 : errors.teamId,
    label: "Vercel Team ID (optional)",
    name: "teamId",
    ref: register
  }), /*#__PURE__*/_react2["default"].createElement(_FormFieldInputText["default"], {
    description: "Enter a valid deploy hook URL to enable manual deploys",
    disabled: formUpdating,
    error: errors === null || errors === void 0 ? void 0 : errors.deployHook,
    label: "Vercel Deploy Hook (optional)",
    name: "deployHook",
    ref: register
  }), /*#__PURE__*/_react2["default"].createElement(_FormFieldInputText["default"], {
    disabled: formUpdating,
    error: errors === null || errors === void 0 ? void 0 : errors.deployLimit,
    label: "Number of deploys to display",
    name: "deployLimit",
    ref: register({
      valueAsNumber: true
    })
  }))));
};

var _default = DialogForm;
exports["default"] = _default;
