// sanity-plugin-dashboard-widget-vercel/lib/components/FormFieldInputText/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ui = require("@sanity/ui");

var _react = _interopRequireWildcard(require("react"));

var _FormFieldInputLabel = _interopRequireDefault(require("../FormFieldInputLabel"));

var FormFieldInputText = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var description = props.description,
      disabled = props.disabled,
      error = props.error,
      label = props.label,
      name = props.name,
      placeholder = props.placeholder,
      value = props.value;
  return /*#__PURE__*/_react["default"].createElement(_ui.Box, null, /*#__PURE__*/_react["default"].createElement(_FormFieldInputLabel["default"], {
    description: description,
    error: error,
    label: label,
    name: name
  }), /*#__PURE__*/_react["default"].createElement(_ui.TextInput, {
    autoComplete: "off",
    autoFocus: true,
    defaultValue: value,
    disabled: disabled,
    id: name,
    name: name,
    placeholder: placeholder,
    ref: ref
  }));
});
var _default = FormFieldInputText;
exports["default"] = _default;
