// sanity-plugin-dashboard-widget-vercel/lib/components/Deployment/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ui = require("@sanity/ui");

var _react = _interopRequireWildcard(require("react"));

var _reactTimeAgo = _interopRequireDefault(require("react-time-ago"));

var _TableCell = _interopRequireDefault(require("../TableCell"));

var _StatusDot = _interopRequireDefault(require("../StatusDot"));

var _icons = require("@sanity/icons");

var Deployment = function Deployment(props) {
  var _deployment$meta, _deployment$meta2, _deployment$alias, _deployment$creator;

  var deployment = props.deployment;
  var date = (0, _react.useRef)(new Date(deployment.created));
  var commitMessage = deployment === null || deployment === void 0 ? void 0 : (_deployment$meta = deployment.meta) === null || _deployment$meta === void 0 ? void 0 : _deployment$meta.githubCommitMessage;
  var commitRef = deployment === null || deployment === void 0 ? void 0 : (_deployment$meta2 = deployment.meta) === null || _deployment$meta2 === void 0 ? void 0 : _deployment$meta2.githubCommitRef;
  var targetUrl = (_deployment$alias = deployment.alias) !== null && _deployment$alias !== void 0 ? _deployment$alias : deployment.url;
  return /*#__PURE__*/_react["default"].createElement("tr", null, /*#__PURE__*/_react["default"].createElement(_TableCell["default"], null, /*#__PURE__*/_react["default"].createElement(_ui.Flex, {
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    display: ['block', 'block', 'block', 'block', 'none'],
    marginRight: 3,
    style: {
      flexShrink: 0
    }
  }, /*#__PURE__*/_react["default"].createElement(_StatusDot["default"], {
    state: deployment.state
  })), targetUrl ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, deployment.alias && /*#__PURE__*/_react["default"].createElement(_icons.LinkIcon, null), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginLeft: deployment.alias ? 1 : 0
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    muted: !(deployment.state === 'READY'),
    size: 1,
    style: {
      textDecoration: deployment.state === 'CANCELED' || deployment.state === 'ERROR' ? 'line-through' : 'normal'
    },
    textOverflow: "ellipsis"
  }, deployment.state === 'READY' ? /*#__PURE__*/_react["default"].createElement("a", {
    href: "https://".concat(targetUrl),
    rel: "noopener noreferrer",
    target: "_blank"
  }, targetUrl) : targetUrl))) : /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1
  }, "Uploading..."))), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], {
    variant: "state"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Flex, {
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(_StatusDot["default"], {
    state: deployment.state
  }), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginLeft: 2
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1
  }, deployment.state.trim().toLowerCase().replace(/^[a-z]/i, function (t) {
    return t.toUpperCase();
  }))))), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], {
    variant: "branch"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 2
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1,
    textOverflow: "ellipsis"
  }, commitRef), commitMessage && /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    muted: true,
    size: 1,
    textOverflow: "ellipsis"
  }, commitMessage))), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], {
    variant: "age"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Flex, {
    align: "center"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1
  }, /*#__PURE__*/_react["default"].createElement(_reactTimeAgo["default"], {
    date: date.current,
    locale: "en-US",
    timeStyle: "mini"
  })))), /*#__PURE__*/_react["default"].createElement(_TableCell["default"], {
    variant: "creator"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Flex, {
    align: "center",
    justify: "center"
  }, /*#__PURE__*/_react["default"].createElement("img", {
    draggable: false,
    src: "https://vercel.com/api/www/avatar/".concat(deployment === null || deployment === void 0 ? void 0 : (_deployment$creator = deployment.creator) === null || _deployment$creator === void 0 ? void 0 : _deployment$creator.uid, "?&s=48"),
    style: {
      borderRadius: '20px',
      height: '20px',
      width: '20px'
    }
  }))));
};

var _default = Deployment;
exports["default"] = _default;
