// sanity-plugin-dashboard-widget-vercel/lib/machines/dialog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _xstate = require("xstate");

/* eslint-disable @typescript-eslint/no-explicit-any */
var dialogMachine = function dialogMachine() {
  return (0, _xstate.Machine)({
    context: {
      editDeploymentTarget: undefined
    },
    initial: 'idle',
    states: {
      idle: {
        entry: (0, _xstate.assign)({
          editDeploymentTarget: function editDeploymentTarget() {
            return undefined;
          }
        }),
        on: {
          CREATE: 'create',
          EDIT: {
            actions: ['setEditDeploymentTarget'],
            target: 'edit'
          }
        }
      },
      edit: {
        on: {
          CLOSE: 'idle'
        }
      },
      create: {
        on: {
          CLOSE: 'idle'
        }
      }
    }
  }, {
    actions: {
      setEditDeploymentTarget: (0, _xstate.assign)(function (_context, event) {
        return {
          editDeploymentTarget: event.deploymentTarget
        };
      })
    }
  });
};

var _default = dialogMachine;
exports["default"] = _default;
