// sanity-plugin-dashboard-widget-vercel/lib/machines/form.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _xstate = require("xstate");

/* eslint-disable */
var formMachine = (0, _xstate.Machine)({
  context: {
    formData: {},
    message: ''
  },
  initial: 'idle',
  states: {
    idle: {
      on: {
        CREATE: {
          actions: ['createDocument'],
          target: 'creating'
        },
        DELETE: {
          actions: ['deleteDocument'],
          target: 'deleting'
        },
        UPDATE: {
          actions: ['updateDocument'],
          target: 'updating'
        }
      }
    },
    creating: {
      invoke: {
        src: 'createDocumentService',
        onDone: {
          target: 'success'
        },
        onError: {
          actions: ['setMessage'],
          target: 'error'
        }
      },
      on: {
        RESOLVE: 'success',
        REJECT: 'error'
      }
    },
    updating: {
      invoke: {
        src: 'updateDocumentService',
        onDone: {
          target: 'success'
        },
        onError: {
          actions: ['setMessage'],
          target: 'error'
        }
      },
      on: {
        RESOLVE: 'success',
        REJECT: 'error'
      }
    },
    deleting: {
      invoke: {
        src: 'deleteDocumentService',
        onDone: {
          target: 'success'
        },
        onError: {
          actions: ['setMessage'],
          target: 'error'
        }
      },
      on: {
        RESOLVE: 'success',
        REJECT: 'error'
      }
    },
    success: {
      invoke: {
        src: 'formSubmittedService'
      }
    },
    error: {}
  }
}, {
  actions: {
    setMessage: (0, _xstate.assign)(function (_context, event) {
      return {
        message: event.data.details.description
      };
    }),
    createDocument: (0, _xstate.assign)(function (_context, event) {
      return {
        formData: event.formData
      };
    }),
    deleteDocument: (0, _xstate.assign)(function (_context, _event) {
      return {// id: event.id,
      };
    }),
    updateDocument: (0, _xstate.assign)(function (_context, event) {
      return {
        formData: event.formData
      };
    })
  }
});
var _default = formMachine;
exports["default"] = _default;
