// sanity-plugin-dashboard-widget-vercel/lib/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Z_INDEX_TOAST_PROVIDER = exports.Z_INDEX_DIALOG = exports.WIDGET_NAME = exports.VERCEL_STATUS_COLORS = exports.DEPLOYMENT_TARGET_DOCUMENT_TYPE = exports.DEBUG_MODE = exports.API_VERSION = exports.API_ENDPOINT_ALIASES = exports.API_ENDPOINT_DEPLOYMENTS = void 0;
// https://vercel.com/docs/platform/limits
var API_ENDPOINT_DEPLOYMENTS = 'https://api.vercel.com/v5/now/deployments';
exports.API_ENDPOINT_DEPLOYMENTS = API_ENDPOINT_DEPLOYMENTS;
var API_ENDPOINT_ALIASES = 'https://api.vercel.com/v3/now/aliases'; // Sanity API version

exports.API_ENDPOINT_ALIASES = API_ENDPOINT_ALIASES;
var API_VERSION = '1';
exports.API_VERSION = API_VERSION;
var DEBUG_MODE = false;
exports.DEBUG_MODE = DEBUG_MODE;
var DEPLOYMENT_TARGET_DOCUMENT_TYPE = 'vercel.deploymentTarget';
exports.DEPLOYMENT_TARGET_DOCUMENT_TYPE = DEPLOYMENT_TARGET_DOCUMENT_TYPE;
var VERCEL_STATUS_COLORS = {
  BUILDING: '#f5a623',
  CANCELED: '#ff0000',
  ERROR: '#ff0000',
  READY: '#50e3c2',
  QUEUED: '#333'
}; // Name displayed in toasts

exports.VERCEL_STATUS_COLORS = VERCEL_STATUS_COLORS;
var WIDGET_NAME = 'Vercel (dashboard)'; // NOTE: Manually set plugin z-index values to be higher than Sanity's header search field
// (which is currently 500202). Also ensure toasts always sit above dialogs.

exports.WIDGET_NAME = WIDGET_NAME;
var Z_INDEX_DIALOG = 600001;
exports.Z_INDEX_DIALOG = Z_INDEX_DIALOG;
var Z_INDEX_TOAST_PROVIDER = 600002;
exports.Z_INDEX_TOAST_PROVIDER = Z_INDEX_TOAST_PROVIDER;
