// sanity-plugin-dashboard-widget-vercel/lib/components/FormFieldInputLabel/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _icons = require("@sanity/icons");

var _ui = require("@sanity/ui");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var StyledErrorOutlineIcon = (0, _styledComponents["default"])(_icons.ErrorOutlineIcon)(function (_ref) {
  var theme = _ref.theme;
  return {
    color: theme.sanity.color.spot.red
  };
});

var FormFieldInputLabel = function FormFieldInputLabel(props) {
  var description = props.description,
      error = props.error,
      label = props.label,
      name = props.name;
  return /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginBottom: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Inline, {
    space: 2
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    as: "label",
    htmlFor: name,
    size: 1,
    weight: "semibold"
  }, label), error && /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1
  }, /*#__PURE__*/_react["default"].createElement(_ui.Tooltip, {
    content: /*#__PURE__*/_react["default"].createElement(_ui.Box, {
      padding: 2
    }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
      muted: true,
      size: 1
    }, /*#__PURE__*/_react["default"].createElement(StyledErrorOutlineIcon, {
      style: {
        marginRight: '0.1em'
      }
    }), error.message)),
    fallbackPlacements: ['top', 'left'],
    placement: "right",
    portal: true
  }, /*#__PURE__*/_react["default"].createElement(StyledErrorOutlineIcon, null)))), description && /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginY: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    htmlFor: name,
    muted: true,
    size: 1
  }, description)));
};

var _default = FormFieldInputLabel;
exports["default"] = _default;
