// sanity-plugin-dashboard-widget-vercel/lib/components/StateDebug/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ui = require("@sanity/ui");

var _react = _interopRequireDefault(require("react"));

var _constants = require("../../constants");

var StateDebug = function StateDebug(props) {
  var name = props.name,
      state = props.state;

  if (!_constants.DEBUG_MODE) {
    return null;
  }

  return /*#__PURE__*/_react["default"].createElement(_ui.Card, {
    scheme: "dark",
    style: {
      backgroundColor: 'rgba(0, 0, 255, 0.9)',
      borderRadius: '3px',
      fontSize: 1,
      fontWeight: 500,
      lineHeight: 'body',
      right: 0,
      opacity: 0.75,
      pointerEvents: 'none',
      position: 'absolute',
      textAlign: 'left',
      top: 0,
      zIndex: 9000
    }
  }, /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    padding: 2
  }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 2
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 0
  }, "Name: ", name), /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 0
  }, "state.value: ", JSON.stringify(state.value)))));
};

var _default = StateDebug;
exports["default"] = _default;
